mlrun/model_monitoring/db/tsdb/tdengine/tdengine_connector.py

```diff
@@ -186,34 +188,55 @@ def get_model_endpoint_real_time_metrics(
     def get_records(
         self,
         table: str,
-        start: str,
-        end: str,
+        start: datetime,
+        end: datetime,
         columns: typing.Optional[list[str]] = None,
         filter_query: str = "",
+        interval: str = "",
+        agg: typing.Optional[list] = None,
+        limit: int = 0,
+        sliding_window: str = "",
         timestamp_column: str = mm_schemas.EventFieldType.TIME,
     ) -> pd.DataFrame:
         """
         Getting records from TSDB data collection.
         :param table:            Either a supertable or a subtable name.
-        :param columns:          Columns to include in the result.
-        :param filter_query:     Optional filter expression as a string. The filter structure depends on the TSDB
-                                 connector type.
         :param start:            The start time of the metrics.
         :param end:              The end time of the metrics.
+        :param columns:          Columns to include in the result.
+        :param filter_query:     Optional filter expression as a string. TDengine supports SQL-like syntax.
+        :param interval:         The interval to aggregate the data by. Note that if interval is provided,
+                                 agg must bg provided as well. Provided as a string in the format of
+                                 '1m', '1h', etc.
+        :param agg:              The aggregation functions to apply on the columns. Note that if agg is provided,
+                                 interval must bg provided as well. Provided as a list of strings in the format of
+                                 ['sum', 'avg', 'count', ...].
+        :param limit:            The maximum number of records to return.
+        :param sliding_window:   The time step for which the time window moves forward. Note that if sliding_window is
```

Comment:
The name `sliding_window` isn't informative enough. Best guess would be that it's the sliding window's size, but even that's wrong according to the description. Maybe `sliding_window_step`?

Reply:
renamed to `sliding_window_step`